1-JavaScript/Week3/homework/ex7-mindPrivacy.js

```diff
@@ -25,10 +25,27 @@ const employeeRecords = [
     salary: 60000,
   },
 ];
-
-function filterPrivateData(/* parameter(s) go here */) {
-  // TODO complete this function
+// i tried 2 ways both give the right output but it doesn't path the npm test any way
+function filterPrivateData(employeeRecords){
+  const newArr = [];
+  for(let i=0;i<employeeRecords.length;i++){
+    newArr.push((({name,occupation,gender})=>({name,occupation,gender}))(employeeRecords[i]));
+  }
+  return newArr
```

Comment:
well done! I am going to make only one point: it is a bit difficult to read. Don't worry  if you use more than one line to solve a problem, it is also very important that everyone can easily understand your code, here my proposal to make it a bit easier to read:

```suggestion
function filterPrivateData(employeeRecords){
  const newArr = [];
  
  for(const employee of employeeRecords){
    const { name, occupation, email } = employee;
    newArr.push({ name, occupation, email });
  }

  return newArr;
}
```

As you can see, it is adding only one more line, but it is a bit clearer what is happening. You could use employeeRecords.forEach as well instead of for..of.